splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptForTriggerMultipleStatements.java

```diff
@@ -0,0 +1,58 @@
+package com.splicemachine.derby.impl.sql.catalog.upgrade;
+
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.sql.dictionary.DataDictionary;
+import com.splicemachine.db.iapi.sql.dictionary.TableDescriptor;
+import com.splicemachine.db.iapi.store.access.ConglomerateController;
+import com.splicemachine.db.iapi.store.access.TransactionController;
+import com.splicemachine.db.impl.sql.catalog.SYSTRIGGERSRowFactory;
+import com.splicemachine.db.impl.sql.catalog.TabInfoImpl;
+import com.splicemachine.derby.impl.sql.catalog.SpliceDataDictionary;
+import com.splicemachine.derby.impl.store.access.hbase.HBaseController;
+import com.splicemachine.utils.SpliceLogUtils;
+
+/**
+ * Created by msirek on 11/5/19.
+ */
+public class UpgradeScriptForTriggerMultipleStatements extends UpgradeScriptBase {
+    public UpgradeScriptForTriggerMultipleStatements(SpliceDataDictionary sdd, TransactionController tc) {
+        super(sdd, tc);
+    }
+
+    @Override
+    protected void upgradeSystemTables() throws StandardException {
+        ConglomerateController heapCC = null;
+        try {
+            TabInfoImpl tII = sdd.getNonCoreTIByNumber(DataDictionary.SYSTRIGGERS_CATALOG_NUM);
+            TableDescriptor td =
+               sdd.getTableDescriptor( tII.getCatalogRowFactory().getCatalogName(),
+                                       sdd.getSystemSchemaDescriptor(), tc );
+            long conglomID = td.getHeapConglomerateId();
+            heapCC=tc.openConglomerate(conglomID,
+                                       false,0,
+                                       TransactionController.MODE_RECORD,
+                                       TransactionController.ISOLATION_REPEATABLE_READ);
+            // If upgrade has already been done, and we somehow got here again by
+            // mistake, don't re-add the multistatement triggers columns to the systriggers
+            // conglomerate descriptor.
+            if (heapCC instanceof HBaseController) {
+                HBaseController hCC = (HBaseController)heapCC;
+                if (hCC.getConglomerate().getFormat_ids().length >=
+                    SYSTRIGGERSRowFactory.SYSTRIGGERS_COLUMN_COUNT) {
```

Comment:
I think we should have a "hardcoded" value here, instead of referring to SYSTRIGGERSRowFactory.SYSTRIGGERS_COLUMN_COUNT (it could be a static value that just copies the current SYSTRIGGERSRowFactory.SYSTRIGGERS_COLUMN_COUNT value)

Otherwise if we add more columns to systriggers this code could be _triggered_ again.

Reply:
That's an excellent point, we hardcode 19 and 20 below anyway, so we can do the same here.
I will also modify it in `UpgradeScriptForTriggerWhenClause.java`